File: FrontEnd/library-app/src/app/shared/models/UserCheckout.ts
import { Book } from "./Book";
import { LibraryUser } from "./LibraryUser";

export interface IUserCheckout{
    userCheckoutId: number;
    bookId: number;
    libraryUserId: number;
    dateCheckedOut: string;
    book: Book;
    libraryUser: LibraryUser;
}

export class UserCheckout implements IUserCheckout{
    userCheckoutId!: number;
    bookId!: number;
    libraryUserId!: number;
    dateCheckedOut!: string;
    book!: Book;
    libraryUser!: LibraryUser;
}

File: BackEnd/LibraryModel/Model/UserCheckout.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LibraryModel.Model
{
    public class UserCheckout
    {
        public int UserCheckoutId {  get; set; }
        public int BookId { get; set; }
        public string? LibraryUserId { get; set; }
        public DateTime DateCheckedOut { get; set; }
        public Book? Book { get; set; }
        public LibraryUser? LibraryUser { get; set; }

    }
}
